### PR TITLE
Feature/offline analytics

### PR DESCRIPTION
**Pull request by thanasio**
rebased to the best of my ability. This version, keeps an offline stack record  of a fixed sized length. When the stack overflows, its bottom recs are deleted and a lost records accumulator is updated. On signal recovery, all events and lost record tracks are uploaded to ga and the local storage is cleared and starts from scratch.

I intend to write some tests at some point as there is quite a bit of functionality there!

### CODE REVIEW

File: app/scripts/decorators/exceptionhandler-decorator.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    $provide.decorator('$exceptionHandler', function($delegate, $injector) {
-      var trackingService;
-      return function(exception, cause) {
-        $delegate(exception, cause);
-        trackingService = trackingService || $injector.get('trackingService');
-        trackingService.tracker.sendException(exception.message, false);
-      };
-    });
-  });
+//angular.module('lmisChromeApp')
```
Comment (tlvince) [inline app/scripts/decorators/exceptionhandler-decorator.js:3]: @thanasio any reason why this was commented?

File: app/scripts/fixtures/analytics_lost_records.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (tlvince) [inline app/scripts/fixtures/analytics_lost_records.json]: Are these empty fixtures necessary?

Comment (thanasio) [inline app/scripts/fixtures/analytics_lost_records.json]: no, just forgot to delete them!

File: app/scripts/services/analytics-sync-service.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+angular.module('lmisChromeApp')
+  .service('analyticsSyncService', function($q, storageService, trackingFactory) {
+    var tracker = trackingFactory.tracker;
+    
+    //syncs clicks events
+    this.syncClicks = function() {
```
Comment (tlvince) [inline app/scripts/services/analytics-sync-service.js]: `syncClicks`, `syncExceptions` and `syncPageViews` are mostly the besides different parameters. Maybe you modularise into a helper for DRY?

File: app/scripts/decorators/exceptionhandler-decorator.js (hunk 1 above)
Comment (thanasio) [inline app/scripts/decorators/exceptionhandler-decorator.js:3]: not really, was aiming to get the version working and passing the tests, then  refactor the factory into the service above.

File: app/scripts/fixtures/analytics_lost_records.json (hunk 2 above)
Comment (thanasio) [inline app/scripts/fixtures/analytics_lost_records.json]: same applies to clicks, pageviews, exceptions.

File: config/production.json [hunk 4]
```diff
@@ -4,6 +4,9 @@
   },
   "analytics": {
     "service": "LoMIS",
-    "propertyID": "UA-52016752-1"
+    "propertyID": "UA-51340003-1",
+    "events_limit": 1024,
```
Comment (thanasio) [inline config/production.json:8]: A note to remember changing the values for the record_limits. That is low for testing purposes. Needs a higher value!

File: app/scripts/services/analytics-sync-service.js (hunk 3 above)
Comment (thanasio) [inline app/scripts/services/analytics-sync-service.js]: Ooops..True...:)Will do this!

**Comment by tlvince** [comment]
Superseded by #477

**Pull request closed without merging**